plugins/woocommerce/client/blocks/assets/js/blocks/product-reviews/inner-blocks/review-date/edit.tsx

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+import { useEntityProp } from '@wordpress/core-data';
+import type { BlockEditProps } from '@wordpress/blocks';
+import {
+	dateI18n,
+	humanTimeDiff,
+	getSettings as getDateSettings,
+} from '@wordpress/date';
+import {
+	InspectorControls,
+	useBlockProps,
+	// @ts-expect-error - Experimental component
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalDateFormatPicker as DateFormatPicker,
+} from '@wordpress/block-editor';
+// Fix dependency group error
+// eslint-disable-next-line @woocommerce/dependency-group
+import { PanelBody, ToggleControl } from '@wordpress/components';
+// eslint-disable-next-line @woocommerce/dependency-group
+import { __, _x } from '@wordpress/i18n';
+
+/**
+ * Renders the `woocommerce/product-review-date` block on the editor.
+ *
+ * @param {Object} props                   React props.
+ * @param {Object} props.setAttributes     Callback for updating block attributes.
+ * @param {Object} props.attributes        Block attributes.
+ * @param {string} props.attributes.format Format of the date.
+ * @param {string} props.attributes.isLink Whether the author name should be linked.
+ * @param {Object} props.context           Inherited context.
+ * @param {string} props.context.commentId The comment ID.
+ *
+ * @return {JSX.Element} React element.
+ */
+export default function Edit( {
+	attributes: { format, isLink },
+	context: { commentId },
+	setAttributes,
+}: BlockEditProps< {
+	format: string;
+	isLink: boolean;
+} > & {
+	context: { commentId: number };
+} ) {
+	const blockProps = useBlockProps();
+	let [ date ] = useEntityProp(
+		'root',
+		'comment',
+		'date',
+		String( commentId )
+	);
```

Comment:
This looks good but note that we can really test this right now as the Comment Template block return nothing when added inside Single Product block. We can revisit and fix it in the PR adding the Review Template block though.